### ISSUE TITLE
Continuous monitoring

### ISSUE THREAD

**Issue by yatli** [sist2app/sist2#240]
Would be cool to integrate fanotify (FYI `fatrace` works pretty well) which generates a list of changes to filesystem entries.
These paths are then deduplicated with tables and logged to a file list, which sist2 can already handle.
One can then setup cron jobs to periodically rotate the changelog into sist2 index, and only run full incremental scans at coarse granularity for consistency.

**Comment by simon987** [comment]
Are you proposing to integrate continuous monitoring within the sist2 codebase? IMO it would make more sense to rotate `fanotify` logs with a python script for example.

In pseudocode:
```
sist2 scan /data -o ./full_index.sist2

while true:
    fanotify | my_rotate_script.py | sist2 --list-file - --incremental ./full_index.sist2 /data -o ./updated_index.sist2
    sist2 index ./updated_index.sist2 &
```

`my_rotate_script.py` would send fanotify lines in batches of 10k files or truncate every 1h etc.

**Comment by yatli** [comment]
Yes I fully agree with you about external mix and match -- not all filesystem X work well with notify tool Y.

**Comment by Rikka93** [comment]
It would be nice if fanotify and some script could be included in the docker image so that continuous monitoring works out of the box for at least ext4 users.

**Comment by shyyio** [comment]
Closing as stale - sist2-admin periodical scan should be extremely fast (roughly as fast as enumerating files with "find")
